### PR TITLE
API call for making a new offer

### PR DESCRIPTION
**Pull request by orbitalturtle**
We add a new API call to allow building BOLT 12 offers! We need this for our integration tests over at lndk.

### CODE REVIEW

**Comment by orbitalturtle** [comment]
Reminder: I need to allow the caller to set a custom amount_msats value.

File: src/node_api.rs [hunk 1]
```diff
@@ -139,6 +143,27 @@ impl Node {
 		}
 	}
 
+	// build an offer to receive payments at this node. path_pubkeys lists the nodes the path will contain,
+	// starting with the introduction node and ending in the destination node (the current node).
+	pub async fn create_offer(
```
Comment (carlaKC) [inline src/node_api.rs:149]: Out of interest - do you think there's _any_ chance of this being added to `ldk_sample` upstream?

File: src/node_api.rs [hunk 2]
```diff
@@ -139,6 +143,27 @@ impl Node {
...
+		&self, path_pubkeys: &[PublicKey], network: Network, msats: u64, quantity: Quantity,
+	) -> Result<Offer, Bolt12SemanticError> {
+		let secp_ctx = Secp256k1::new();
+		let path =
+			BlindedPath::new_for_message(path_pubkeys, &*self.keys_manager, &secp_ctx).unwrap();
+		let (pubkey, _) = self.get_node_info();
+
+		let expiration = SystemTime::now() + Duration::from_secs(24 * 60 * 60);
```
Comment (carlaKC) [inline src/node_api.rs]: _Might_ want to make this configurable? Can see us wanting to test expiry in the future?

File: src/onion.rs [hunk 3]
```diff
@@ -46,8 +46,7 @@ impl CustomOnionMessageHandler for OnionMessageHandler {
 		Ok(Some(UserOnionMessageContents { tlv_type: message_type, data: buf.to_vec() }))
 	}
 
-        // We don't need this for our purposes yet.
         fn release_pending_custom_messages(&self) -> Vec<PendingOnionMessage<Self::CustomMessage>> {
-		unimplemented!();
+		vec![]
```
Comment (carlaKC) [inline src/onion.rs:56]: Why this change?

**Review by carlaKC — COMMENTED** [review]
Driveby comments!

File: src/node_api.rs (hunk 1 above)
Comment (orbitalturtle) [reply src/node_api.rs:149]: Good question! I'll ask

File: src/onion.rs (hunk 3 above)
Comment (orbitalturtle) [reply src/onion.rs:56]: Was getting an error with unimplemented!(), and saw that vec![] seems to be the de facto implementation for now in rust-lightning's integration tests. I don't *think* we really need a full implementation of release_pending_custom_messages for our tests, at least right now, but I should probably dive in and double check that

**Comment by orbitalturtle** [comment]
Merging this since my next PR depends on this.
